### DISCUSSION TITLE
Plotting state-dependent gamma distribution with density = Inf at x = 0

### DISCUSSION THREAD

**Discussion by DEHewitt** [bmcclintock/momentuHMM#161] [Q&A]
Hi @bmcclintock, this question was previously asked in response to #93, but I am moving it here in case the notification doesn't come through (as that is marked 'Answered') and to avoid notifying the original author of that discussion.

I am using the code in #93 to plot state-dependent distributions from a 2-state HMM fitted to step length (zero-inflated gamma), turning angle (wrapped Cauchy) and acceleration (zero-inflated gamma) data. However, I am getting  `Inf` density at `x = 0` for both step length distributions and the state 2 distribution for acceleration (plot below) which don't look like gamma distributions to me. My best guess is this is because of the relatively high SD in these states (summary below), which imply values < 0. Is this an issue? And, if so is it evidence to include more states, or (if biologically sensible) are inferences based on this model valid?
```
step parameters:
----------------
            state 1      state 2
mean     0.80250291 1.193297e+01
sd       0.92272226 1.487667e+01
zeromass 0.01372569 5.924174e-07

angle parameters:
-----------------
                state 1   state 2
mean          0.0000000 0.0000000
concentration 0.4182714 0.4879156

accel parameters:
-------------------------
             state 1     state 2
mean     0.041852155 0.471148341
sd       0.015188037 0.513780719
zeromass 0.002279825 0.007687249
```

![image](https://user-images.githubusercontent.com/54158290/211976265-cfe9e21c-a556-42d5-bb73-a2ea89d7fbdb.png)

**Comment by bmcclintock** [comment]
I'm not sure I understand your question, but the gamma distribution is only defined for `x > 0`. The zero-inflated gamma distribution is a mixture of two distributions, one for `x = 0` (with mixture probability `zeromass`) and one for `x > 0` (with mixture probability `1-zeromass`). If you wish to plot the density for `x > 0`, you should not be including `x = 0`.

**Reply by DEHewitt** [reply]
Ok, so the right approach would be to plot the estimated gamma distribution and report the associated mixture probability (`zeromass`)?

I guess I am still a little confused about why when I plot my fitted model (`m`) using `plot(m)` the below is returned, which appears to have some density at `step = 0` and looks the same as the plots I produced (except for truncating the x-axis/binning observed data differently).

![image](https://user-images.githubusercontent.com/54158290/213589987-4395c428-8393-4e9f-86a9-11178614a98f.png)

 I also note that in the code for #93 a sequence of `x` values are generated which include `0` (i.e., `x <- seq(0, 2000, length=1000)`), however those distributions (correctly have `density = 0` at `x = 0`). Could my particular case be because I have the following mean/sd:

```
step1 <- m$CIreal$step$est[1]  #0.4417
sd1 <- m$CIreal$step$est[2]    #5.6714
    
step2 <- m$CIreal$step$est[4]  #0.0420
sd2 <- m$CIreal$step$est[5]    #1.1039
```
and when I convert these to `shape`/`scale` for downstream use in `dgamma()`, I get `shape < 1`:
```
shape1 <- step1^2/sd1^2 #0.7596
scale1 <- sd1^2/step1   #6.507

shape2 <- step2^2/sd2^2 #0.4431
scale2 <- sd2/step2     #1.6583
```

My distributions don't look too dissimilar from the [pink gamma distribution here](https://en.wikipedia.org/wiki/Gamma_distribution#/media/File:Gamma_distribution_pdf.svg) with `shape` (k) = 0.5 and `scale` (θ) = 1.

I appreciate you taking the time to answer my questions.
